db/import_column_family_job.cc

```diff
@@ -29,33 +29,76 @@ namespace ROCKSDB_NAMESPACE {
 Status ImportColumnFamilyJob::Prepare(uint64_t next_file_number,
                                       SuperVersion* sv) {
   Status status;
+  std::vector<ColumnFamilyIngestFileInfo> cf_ingest_infos;
+  for (const auto& metadata_per_cf : metadatas_) {
+    // Read the information of files we are importing
+    ColumnFamilyIngestFileInfo cf_file_info;
+    InternalKey smallest, largest;
+    int num_files = 0;
+    for (size_t i = 0; i < metadata_per_cf.size(); i++) {
+      auto file_metadata = metadata_per_cf[i];
+      const auto file_path = file_metadata.db_path + "/" + file_metadata.name;
+      IngestedFileInfo file_to_import;
+      status = GetIngestedFileInfo(file_path, next_file_number++, sv,
+                                   file_metadata, &file_to_import);
+      if (!status.ok()) {
+        return status;
+      }
 
-  // Read the information of files we are importing
-  for (const auto& file_metadata : metadata_) {
-    const auto file_path = file_metadata.db_path + "/" + file_metadata.name;
-    IngestedFileInfo file_to_import;
-    status = GetIngestedFileInfo(file_path, next_file_number++, sv,
-                                 file_metadata, &file_to_import);
-    if (!status.ok()) {
-      return status;
-    }
-    files_to_import_.push_back(file_to_import);
-  }
+      if (file_to_import.num_entries == 0) {
+        status = Status::InvalidArgument("File contain no entries");
+        return status;
+      }
 
-  auto num_files = files_to_import_.size();
-  if (num_files == 0) {
-    status = Status::InvalidArgument("The list of files is empty");
-    return status;
-  }
+      if (!file_to_import.smallest_internal_key.Valid() ||
+          !file_to_import.largest_internal_key.Valid()) {
+        status = Status::Corruption("File has corrupted keys");
+        return status;
+      }
 
-  for (const auto& f : files_to_import_) {
-    if (f.num_entries == 0) {
-      status = Status::InvalidArgument("File contain no entries");
+      files_to_import_.push_back(file_to_import);
+      metadata_.push_back(file_metadata);
```

Comment:
It feels like a lot of copy of file_metadata. We can probably get rid of `metadata_` and use `metadatas_` directly (this may require adapting `file_to_import` to a 2d vector).